### PR TITLE
Wildcard V2: `<Panel />` Implementation

### PR DESCRIPTION
**Pull request by gitstart-sourcegraph**
## Description

Implement `Panel` component in `wildcard` package

## Refs
- [SourceGraph Issue](https://github.com/sourcegraph/sourcegraph/issues/27709)
- [GitStart ticket](https://app.gitstart.com/clients/sourcegraph/tickets/SG-27709)

### CODE REVIEW

**Review by valerybugakov — COMMENTED** [review]
@oleggromov, could you take a look when you have a chance?

**Comment by gitstart-sourcegraph** [comment]
Hello @oleggromov,
Kindly help take a look at changes here.

**Comment by umpox** [comment]
@gitstart-sourcegraph I've just noticed that we also have the `Resizable` component in the codebase. Do you think we could (in a follow-up PR) implement the same logic? It is quite similar to this implementation, but it is an old class component so it would be good to move logic into this hook.

The additional features I see from a quick look:
- [Size changes are debounced for improved performance](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@contractors/SG-27709/-/blob/client/shared/src/components/Resizable.tsx?utm_source=VSCode-1.4.0#L91:49)
- [Holding down the shift key will let you resize in increments](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@contractors/SG-27709/-/blob/client/shared/src/components/Resizable.tsx#L143:25)
- [It supports `right` and `left` alignment too](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@contractors/SG-27709/-/blob/client/shared/src/components/Resizable.tsx#L16:14). Apologies I said I didn't think we needed this before, but it looks like it'll be worthwhile to add now.

If we can implement the same features for this `Panel` component we can remove `Resizable` when we do the migration for this work.

**Review by umpox — APPROVED** [review]
Looks great, thanks! I'll prep the `Panel` migration issue

File: client/wildcard/src/components/Panel/useResizablePanel.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+import { Subject, Subscription } from 'rxjs'
+import { debounceTime, distinctUntilChanged } from 'rxjs/operators'
+
+import { PANEL_POSITIONS } from './constants'
+
+const STORAGE_KEY_PREFIX = 'ResizePanel:'
+
+const savePanelSize = (storageKey: UseResizablePanelParameters['storageKey'], size: number): void => {
+    if (storageKey) {
+        localStorage.setItem(`${STORAGE_KEY_PREFIX}${storageKey}`, String(size))
```
Comment (limitedmage) [inline client/wildcard/src/components/Panel/useResizablePanel.ts:11]: Instead of using localStorage here, could we use temporary settings instead?

**Pull request merged**